test/rules/cascade_invocations.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N cascade_invocations`
+
+void noCascade() {
+  List<int> list = [];
+  list.removeWhere((i) => i % 5 == 0); // LINT
+  list.clear(); // LINT
+}
+
+void noCascadeIntermediate() {
+  List<int> list = [];
+  print(list);
+  list.removeWhere((i) => i % 5 == 0);
+  list.clear(); // LINT
+}
+
+class HasMethodWithNoCascade {
+  List<int> list = [];
+
+  void noCascade() {
+    list.removeWhere((i) => i % 5 == 0);
+    list.clear(); // LINT
+  }
+}
```

Comment:
You need to test negative cases, such as:
```
foo.m();
bar.m();
```
or
```
String s = '';
x.m();
```
where it matches the lexical pattern but doesn't refer to the same variable. You should also test that it work correctly when the invocation is nested in a statement or expression:
```
if (x.m() != null) ...
```